metpy/calc/thermo.py

```diff
@@ -1037,4 +1037,64 @@ def most_unstable_parcel(p, temperature, dewpoint, heights=None,
                                            depth=depth, heights=heights)
     theta_e = equivalent_potential_temperature(p_layer, T_layer)
     max_idx = np.argmax(theta_e)
-    return p_layer[max_idx], T_layer[max_idx], Td_layer[max_idx]
+    return p_layer[max_idx], T_layer[max_idx], Td_layer[max_idx], max_idx
+
+
+@exporter.export
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def mucape_mucin(pressure, temperature, dewpoint, **kwargs):
+    r"""Calculate MUCAPE and MUCIN.
+
+    Calculate the convective available potential energy (CAPE) and convective inhibition (CIN)
+    of a given upper air profile and most unstable parcel path. CIN is integrated between the
+    surface and LFC, CAPE is integrated between the LFC and EL (or top of sounding).
+    Intersection points of the measured temperature profile and parcel profile are linearly
+    interpolated.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        The atmospheric pressure level(s) of interest. The first entry should be the starting
+        point pressure.
+    temperature : `pint.Quantity`
+        The sounding temperature
+    dewpoint : `pint.Quantity`
+        The starting dew point
+
+    Returns
+    -------
+    `pint.Quantity`
+        Convective available potential energy (CAPE) for the most unstable parcel.
+    `pint.Quantity`
+        Convective inhibition (CIN) for the most unstable parcel.
+
+    Notes
+    -----
+    Formula adopted from [Hobbs1977]_.
+
+    .. math:: \text{CAPE} = -R_d \int_{LFC}^{EL} (T_{parcel} - T_{env}) d\text{ln}(p)
+
+    .. math:: \text{CIN} = -R_d \int_{SFC}^{LFC} (T_{parcel} - T_{env}) d\text{ln}(p)
+
+
+    * :math:`CAPE` Convective available potential energy
+    * :math:`CIN` Convective inhibition
+    * :math:`LFC` Pressure of the level of free convection
+    * :math:`EL` Pressure of the equilibrium level
+    * :math:`SFC` Level of the surface or beginning of parcel path
+    * :math:`R_d` Gas constant
+    * :math:`g` Gravitational acceleration
+    * :math:`T_{parcel}` Parcel temperature
+    * :math:`T_{env}` Environment temperature
+    * :math:`p` Atmospheric pressure
+
+    See Also
+    --------
+    lfc, el
+
+    """
+    parcel = most_unstable_parcel(pressure, temperature, dewpoint, **kwargs)
+    pres = pressure[parcel[-1]:]
```

Comment:
Not following at all here. `parcel[-1]` is the dew point of the parcel, so how is that being used as an index here? Are you using the implantation of `most_unstable_parcel` that made it into master?